akirachix/Magezi-Frontend
# added boiler plate

# Engineers Checklist
This PR introduces a new front-end boilerplate for the Shawazi Progressive Web App (PWA), which outlines the need for a standardized project structure. The primary motivation behind this change is to enhance development efficiency and maintainability, ensuring a cohesive starting point for future features.

## What this PR does?
This PR introduces a new front-end boilerplate for the Shawazi Progressive Web App (PWA), which outlines the need for a standardized project structure. The primary motivation behind this change is to enhance development efficiency and maintainability, ensuring a cohesive starting point for future features.

* What is this?
This PR provides a boilerplate structure for the Shawazi PWA, including essential components, routing, state management, and styling conventions.

* Why this?
Created a modular directory structure for components, services, and assets.
Integrated state management with Redux (or any state management library you’re using).
Set up routing using React Router (or equivalent).
Implemented basic theming with CSS-in-JS

* How are you doing this? 
Modularized component architecture.
Added routing capabilities.
Integrated state management.
Established basic styling and theming conventions.

## Type of change

Specify the kind of PR/MR

Please delete options that are not relevant.


## How can it be tested?
Clone the repository and checkout to this branch.
Install dependencies using npm install.
Run the development server with npm start or yarn start.
Open your browser and navigate to http://localhost:3000.
Verify that the boilerplate components render correctly.
Check the routing by navigating through the defined paths.
Test state management by performing actions that change the state and verify updates in the UI.

# Where is it documented?
[https://nakatozainamusana.atlassian.net/jira/software/projects/SCRUM/boards/4/backlog?selectedIssue=SCRUM-52](url)
